jenkins/opensearch-dashboards/Jenkinsfile

```diff
@@ -59,14 +112,21 @@ pipeline {
     }
 }
 
-void build() {
+void build(platform, architecture) {
+    git url: 'https://github.com/opensearch-project/opensearch-build.git', branch: 'main'
+
+    sh "./build.sh manifests/$INPUT_MANIFEST -p $platform -a $architecture"
+}
+
+void assemble() {
     git url: 'https://github.com/opensearch-project/opensearch-build.git', branch: 'main'
 
-    sh "./build.sh manifests/$INPUT_MANIFEST"
     sh './assemble.sh artifacts/manifest.yml'
+}
 
+void publish() {
     script { manifest = readYaml(file: 'artifacts/manifest.yml') }
-    def artifactPath = "${manifest.build.version}/${BUILD_NUMBER}/${manifest.build.architecture}";
+    def artifactPath = "${manifest.build.version}/${BUILD_NUMBER}/${manifest.build.platform}-${manifest.build.architecture}";
```

Comment:
highlighting updated s3 path.

Reply:
You've got a race with https://github.com/opensearch-project/opensearch-build/pull/765 which cleans this all up